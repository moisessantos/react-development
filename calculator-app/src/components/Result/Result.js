import React from 'react';
import PropTypes from 'prop-types';

const Result = ({ children, ...otherProps }) => (
  <div className="result" {...otherProps}>
    <p>{children}</p>
  </div>
);

Result.defaultProps = {
  children: '',
};

Result.propTypes = {
  children: PropTypes.string,
};

export default Result;
