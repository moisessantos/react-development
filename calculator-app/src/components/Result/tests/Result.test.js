import React from 'react';
import { shallow } from 'enzyme';
import Result from '..';

describe('Result tests', () => {
    it('should render as expected', () => {
        const wrapper = shallow(<Result>500</Result>)
        expect(wrapper).toMatchSnapshot();
    });
})