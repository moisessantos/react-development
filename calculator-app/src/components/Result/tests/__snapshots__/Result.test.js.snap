// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result tests should render as expected 1`] = `
<div
  className="result"
>
  <p>
    500
  </p>
</div>
`;
